### PR TITLE
use canonical filename of RTD configuration and enable `nitpicky` mode

### PR DESCRIPTION
**Pull request by zacharyburnett**
### Description

update the name of the ReadTheDocs configuration file from `.readthedocs.yml` to `.readthedocs.yaml` and enable `nitpicky` to validate intra-doc links

### CODE REVIEW

**Comment by zacharyburnett** [comment]
the RTD build ran into a few errors / warnings: https://readthedocs.org/projects/synphot/builds/18123748/

**Comment by pllim** [comment]
I would approve if you also fix the 23 warnings. Thanks! 😉

**Comment by pllim** [comment]
One does not simply turn on nitpicky.

File: synphot/spectrum.py [hunk 1]
```diff
@@ -1799,7 +1799,7 @@ def equivwidth(self, **kwargs):
         Parameters
         ----------
         kwargs : dict
-            See :meth:`integrate`.
+            See :meth:``integrate``.
```
Comment (pllim) [inline synphot/spectrum.py:1802]:
```suggestion
            See :meth:`~synphot.spectrum.BaseSpectrum.integrate`.
```

**Review by pllim — APPROVED** [review]
Thanks!

**Pull request merged**